### PR TITLE
revive new message queue integration

### PR DESCRIPTION
**Pull request by kianenigma**
Just bringing https://github.com/paritytech/cumulus/pull/2157 up to speed with master, and fixing some tests helping me understand. 

up to @ggwpez if this should be merged or not.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2926811

**Comment by ggwpez** [comment]
I have a hard time seeing the actual diff, but it compiles :smile:  
So lets merge it once CI is greenish.

File: xcm/xcm-emulator/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,954 @@
...
+								msg_dedup.dedup();
+
+								let msgs = msg_dedup.clone().into_iter().filter(|m| {
+									!$crate::DMP_DONE.with(|b| b.borrow_mut().get_mut(stringify!($name)).unwrap_or(&mut $crate::VecDeque::new()).contains(&(to_para_id, m.0, m.1.clone())))
+								}).collect::<Vec<(RelayChainBlockNumber, Vec<u8>)>>();
+								if msgs.len() != 0 {
+
+									// #[doc = stringify!($parachain)]
+									// fn todo() {} // CI-FAIL
+									// CI-FAIL: based on the PR, it seems like the right entry point for handling dmp messages is now <parachain_system::DmpQueue> type.
```
[the remark this answers is not in the document]
Comment (kianenigma) [reply xcm/xcm-emulator/src/lib.rs:778]: and I am trying to learn the right change from looking at `enqueue_inbound_downward_messages` in `parachain_system`.

[the remark this answers is not in the document]
Comment (ggwpez) [reply xcm/xcm-emulator/src/lib.rs:778]: Its actually only reacting to `FAIL-CI`, but you can extend the magic words in `scripts/ci/gitlab/pipeline/test.yml`.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2928163

**Comment by kianenigma** [comment]
@ggwpez I merged master into this again. I have a few points of question that are marked with TODO, mainly about some tests in xmc-emulator. 

Given that this is hard to maintain, if you intended to merge this, perhaps we should do it sooner. Happy to drop this as well, it was helpful for me to learn a bit of this code nonetheless. 

As of the last commit, everything including tests build. Not sure if all of them pass though 🙈

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2940180

**Comment by ggwpez** [comment]
I AFAIK the `xcm-emulator` stuff is optional. We could also do it afterwards.  
I originally did it in Polkadot just as a smoke screen test.  
Anywas let me see what still fails...

**Comment by kianenigma** [comment]
I think all CI failures are due to warning. Locally `cargo test --all` passes entirely.

**Pull request merged**